### PR TITLE
[cli11] Update to 2.1.0

### PR DESCRIPTION
**Pull request by chausner**
**Describe the pull request**

- #### What does your PR fix?  
  Updates cli11 to 2.1.0

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  As before, Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

File: ports/cli11/portfile.cmake [hunk 1]
```diff
@@ -16,9 +16,9 @@ vcpkg_cmake_configure(
 
 vcpkg_cmake_install()
 
-vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/CLI11)
+vcpkg_cmake_config_fixup(CONFIG_PATH share/cmake/CLI11)
 vcpkg_fixup_pkgconfig()
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")
```
Comment (chausner) [inline ports/cli11/portfile.cmake:21]: These changes are needed due to https://github.com/CLIUtils/CLI11/pull/635.

File: ports/cli11/vcpkg.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "cli11",
-  "version-semver": "2.0.0",
+  "version-semver": "2.1.0",
```
Comment (JonLiu1993) [inline ports/cli11/vcpkg.json:3]: @chausner ,Could you please confirm whether the library officially adheres to semantic versioning or not?

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/cli11/vcpkg.json (hunk 2 above)
Comment (chausner) [reply ports/cli11/vcpkg.json:3]: According to this older comment, at least in the past the project was not adhering to semantic versioning: https://github.com/CLIUtils/CLI11/pull/340#issuecomment-569962975

Comment (chausner) [reply ports/cli11/vcpkg.json]: I changed it to `"version"`.

**Comment by chausner** [comment]
@JonLiu1993 Can you please try rerunning CI? I think the CI error was just a temporary issue.

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**